controllers/componentdetectionquery_controller.go

```diff
@@ -67,7 +66,11 @@ type ComponentDetectionQueryReconciler struct {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.2/pkg/reconcile
 func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := r.Log.WithValues("ComponentDetectionQuery", req.NamespacedName).WithValues("clusterName", req.ClusterName)
-	ctx = kcpclient.WithCluster(ctx, logicalcluster.New(req.ClusterName))
+
+	// if we're running on kcp, we need to include workspace in context
+	if req.ClusterName != "" {
+		ctx = logicalcluster.WithCluster(context.TODO(), logicalcluster.New(req.ClusterName))
```

Comment:
```suggestion
		ctx = logicalcluster.WithCluster(ctx, logicalcluster.New(req.ClusterName))
```
should be?

Reply:
Oof yes, that was some testing that was not supposed to be committed